diff --git a/libs/ui/input/kis_input_manager.cpp b/libs/ui/input/kis_input_manager.cpp
index 79c3604..4dfb58a 100644
--- a/libs/ui/input/kis_input_manager.cpp
+++ b/libs/ui/input/kis_input_manager.cpp
@@ -561,16 +561,6 @@ bool KisInputManager::eventFilterImpl(QEvent * event)
         //Clear all state so we don't have half-matched shortcuts dangling around.
         d->matcher.reinitialize();
 
-    { // Emulate pressing of the key that are already pressed
-        KisExtendedModifiersMapper mapper;
-
-        Qt::KeyboardModifiers modifiers = mapper.queryStandardModifiers();
-        Q_FOREACH (Qt::Key key, mapper.queryExtendedModifiers()) {
-            QKeyEvent kevent(QEvent::ShortcutOverride, key, modifiers);
-            eventFilterImpl(&kevent);
-        }
-    }
-
         d->allowMouseEvents();
         break;
 
